CONTRIBUTING.markdown

```diff
@@ -1,114 +1,171 @@
+
+For more information, visit the links listed in the Additional Resources section, below.
+
+# Building Liferay Portal
+
+System Requirements:
+* Quad Core i7 Processor
+* 8 GB Memory
+* JDK 8
+* Apache Ant
+* Gradle
+
+The first step to contributing to Liferay Portal is to clone the Liferay Central (**liferay-portal**) repo from Github and build the platform using Apache Ant.  All changes from the Sub Repos are synced to Central so no Sub Repos need to be involved in the build process.  
+
+At the time of this writing the Liferay Central repo weighs in a 12 GB’s.  In order to cut down on the download size the repo can cloned with the options of leaving out commit history and selecting only one branch.
+
+## Build from Central Repo:
+
+To build Liferay Portal from source do the following:
+
+* Fork Liferay Portal Central Repo on Github - [https://github.com/liferay/liferay-portal](https://github.com/liferay/liferay-portal)
+
+* Clone the forked repo’s master branch (or preferred branch) with no commit history:
+
+      git clone https://github.com/<github username>/liferay-portal --branch master --single-branch --depth 1
+
+* Add main liferay-portal repo as an upstream for fetching changes:
+
+      git remote add upstream https://github.com/liferay/liferay-portal
+
+* Build master branch from source for testing customizations:
+
+	    cd liferay-portal
+          ant all
+
+* Test the newly built bundle:
+
+      cd ../bundles
+      ./tomcat-8.0.32/bin/startup.sh
+
+## Building a Module in Central
+
+In order to build a module that is maintained from within Central (aka it isn’t maintained in a Sub Repo as of yet) it will have to be built from within the Central repo.  The good news is that since it’s a module the whole platform does not have to be recompiled.  To build a module located within central do the following:
+
+* The platform must be completely compiled once using the method above.  This is due to the fact that the initial build initializes the build environment.
+
+* To build a module located in: **liferay-portal/modules/apps/collaboration/comment/comment-web** run: 
+
+      cd liferay-portal/modules
+      ../gradlew :apps:collaboration:comment:comment-web:deploy
+
+* If all goes well the following should appear in the bundle logs:
+
+      20:31:55,899 INFO  [fileinstall-/Users/jamie/liferay/repos/contrib/bundles/osgi/modules][BundleStartStopLogger:38] STOPPED com.liferay.comment.web_1.1.8 [243]
+
+      20:31:56,038 INFO  [Refresh Thread: Equinox Container: 208f839d-2b6b-0017-1f31-cc9c6b1fc91b][BundleStartStopLogger:35] STARTED com.liferay.comment.web_1.1.8 [243]
+
+## Build from Sub Repo:
```

Comment:
If you would still prefer to keep this section, I think we need to explain how Sub Repos work and why we are using them at Liferay. 

I also think that working from a Sub Repo should be optional, to avoid having to clone liferay-portal.